tsnet/tsnet.go

```diff
@@ -959,6 +959,11 @@ func (s *Server) ListenTLS(network, addr string) (net.Listener, error) {
 	if err != nil {
 		return nil, err
 	}
+	if len(st.CertDomains) != 0 {
+		if st.CurrentTailnet.MagicDNSEnabled == false {
+			return nil, errors.New("tsnet: you must enable MagicDNS in the in the DNS page of the admin console. See https://login.tailscale.com/admin/dns")
+		}
+	}
```

Comment:
The HTTPS setup docs mention enabling MagicDNS so it can point there. If MagicDNS is disabled, the control server won't send any `st.CertDomains` so the length will be 0 in the MagicDNS disabled case too.

```suggestion
	if !st.CurrentTailnet.MagicDNSEnabled {
		return nil, errors.New("tsnet: you must enable MagicDNS in the in the DNS page of the admin panel to proceed. See https://tailscale.com/s/https")
	}
```